weather.html

```diff
@@ -0,0 +1,118 @@
+            return {latitude: latitude, longitude: longitude};
+        } else {
+            throw new Error(`No coordinates found for ${city}!`);
+        }
+    }
+
+    async function getWeatherForCoordinates(coordinates) {
+        let result = await fetch(`https://api.open-meteo.com/v1/forecast?latitude=${coordinates.latitude}&longitude=${coordinates.longitude}&forecast_days=1&daily=temperature_2m_min&daily=temperature_2m_max&timezone=CET`);
+        if (result.ok) {
+            let result_json = await result.json();
+            let min = result_json.daily?.temperature_2m_min?.[0];
+            let max = result_json.daily?.temperature_2m_max?.[0];
+            return {min: min, max: max};
+        } else {
+            throw new Error(`No weather info found for (${coordinates.latitude}, ${coordinates.longitude})!`);
+        }
+    }
+
+    function displayErrorMessage(error) {
+        let error_message_paragraph = document.createElement("p");
+        error_message_paragraph.textContent = error;
+        document.getElementById("error-messages").appendChild(error_message_paragraph);
+    }
+
+    function eraseErrorMessages() {
+        let input = document.getElementById("error-messages");
+        input.innerHTML = "";
+    }
+
+    function displayWeather(cities, cities_weather) {
+        for (let i = 0; i < cities_weather.length; i++) {
+            let city = cities?.[i];
+            let city_weather = cities_weather?.[i];
+            let error_message_paragraph = document.createElement("p");
+            error_message_paragraph.textContent = `${city}: max: ${city_weather.max} min: ${city_weather.min}`;
+            document.getElementById("weather-forecast").appendChild(error_message_paragraph);
+        }
+    }
+
+    function eraseWeather() {
+        let input = document.getElementById("weather-forecast");
+        input.innerHTML = "";
+    }
+
+    async function actionGetWeather() {
+        eraseErrorMessages();
+        eraseWeather();
+
+        let cities = getCitiesFromInput();
+        console.log(cities);
+        if (cities.length === 0 || cities.length > 3) {
+            displayErrorMessage("Your input was parsed as either 0 or more than 3 cities!");
+            return;
+        }
+        for (let i = 0; i < cities.length; i++) {
+            let city = cities[i];
+            if (city.length === 0) {
+                displayErrorMessage("One of the strings parsed from your input has length zero!");
+                return;
+            }
+        }
+
+        let cities_coordinates = cities.map(city => getCityCoordinates(city));
+        cities_coordinates = await Promise.allSettled(cities_coordinates).then(result => result);
+        for (let i = 0; i < cities_coordinates.length; i++) {
+            if (cities_coordinates[i].status === "rejected") {
+                displayErrorMessage(`Coordinates for city ${cities[i]} could not be found!`);
```

Comment:
Tady je trochu skoda/nestastne, ze pri chybe i jen jednoho mesta ze 3 nepokracujes dal s tim, co funguje. Jinymi slovy zobrazit uzivateli napr. alespon zbyla mesta, pro ktera se podarilo ziskat souradnice. Nyni se to pri chybe cele "ukonci". A podobne je ten samy stav pri vypisovani pocasi. Asi bych se vzdy snazil uzivateli alespon trochu vyjit vstric